src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyOfState = { ...state };
+  const arrayOfCopies = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyOfState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key in copyOfState) {
+          if (action.keysToRemove.includes(key)) {
+            delete copyOfState[key];
+          }
+        }
```

Comment:
Instead of iterating through all keys in copyOfState, you should iterate through the keys in action.keysToRemove.
Refactor this code without `if` condition and with `for of`
Let's move this code to separated function outside main function `transformStateWithClones`